app/addons/documents/resources.js

```diff
@@ -120,31 +118,26 @@ Documents.BulkDeleteDocCollection = FauxtonAPI.Collection.extend({
   },
 
   bulkDelete: function () {
-    var payload = this.createPayload(this.toJSON()),
-        promise = FauxtonAPI.Deferred(),
-        that = this;
-
-    $.ajax({
-      type: 'POST',
-      url: this.url(),
-      contentType: 'application/json',
-      dataType: 'json',
-      data: JSON.stringify(payload),
-    }).then(function (res) {
-      that.handleResponse(res, promise);
-    }).fail(function () {
-      var ids = _.reduce(that.toArray(), function (acc, doc) {
-        acc.push(doc.id);
-        return acc;
-      }, []);
-      that.trigger('error', ids);
-      promise.reject(ids);
+    const payload = this.createPayload(this.toJSON());
+    const promise = new FauxtonAPI.Promise((resolve, reject) => {
+      post(this.url(), payload).then(res => {
+        if (res.error) {
```

Comment:
Do you think throwing our own error is the best way of handling this? Could we not just handle it here where we know the error has happened?

Reply:
The issue is unlike `$.ajax()`, `fetch` doesn't raise an error for 4xx, 5xx codes. I could handle errors inside `then()` as you suggested, but I should also add a `catch()` to handle `fetch()` errors as well (e.g. connection lost).
Considering both types of errors are handled pretty much in the same way, I found the easiest is to raise an Error and keep error handling inside `catch()` because it seems more logical.